indico/modules/events/timetable/controllers/legacy.py

```diff
@@ -16,16 +16,62 @@
 
 from __future__ import unicode_literals
 
+from collections import Counter
+
 import dateutil.parser
 from flask import request, jsonify
 from werkzeug.exceptions import BadRequest
 
 from indico.modules.events.contributions import Contribution
+from indico.modules.events.contributions.operations import create_contribution
 from indico.modules.events.sessions.models.sessions import Session
 from indico.modules.events.timetable.controllers import RHManageTimetableBase
+from indico.modules.events.timetable.forms import BreakEntryForm, ContributionEntryForm
 from indico.modules.events.timetable.legacy import serialize_contribution, serialize_entry_update
-from indico.modules.events.timetable.operations import schedule_contribution
+from indico.modules.events.timetable.models.breaks import Break
+from indico.modules.events.timetable.operations import create_break_entry, schedule_contribution
 from indico.modules.events.timetable.util import find_earliest_gap
+from indico.modules.events.util import get_random_color
+from indico.web.forms.base import FormDefaults
+from indico.web.util import jsonify_data, jsonify_form
+
+
+class RHLegacyTimetableAddBreak(RHManageTimetableBase):
+    def _checkParams(self, params):
+        RHManageTimetableBase._checkParams(self, params)
+        self.day = dateutil.parser.parse(request.args['day']).date()
+
+    def _process(self):
+        inherited_location = self.event_new.location_data
+        inherited_location['inheriting'] = True
+        breaks = Break.query.filter(Break.timetable_entry.has(event_new=self.event_new)).all()
+        common_colors = Counter()
+        for break_ in breaks:
+            common_colors[break_.colors] += 1
+        colors = common_colors.most_common()[0][0] if common_colors.most_common() else get_random_color(self.event_new)
```

Comment:
`most_common(1)` since you only care about the most common element, not any less common ones (and i'd assign it instead of calling it twice)